tensorflow/compiler/mlir/xla/tests/xla-hlo-fusion.mlir

```diff
@@ -0,0 +1,69 @@
+// RUN: tf-opt %s -xla-hlo-fusion -split-input-file | FileCheck %s --dump-input-on-failure
+
+// CHECK-LABEL: func @multi_outputs_same
+func @multi_outputs_same(%arg0: tensor<?x?xf32>, %arg1: tensor<?x?xf32>) -> (tensor<?x?xf32>, tensor<?x?xf32>) {
+  %0 = "xla_hlo.add"(%arg0, %arg1) : (tensor<?x?xf32>, tensor<?x?xf32>) -> tensor<?x?xf32>
+  %1 = "xla_hlo.subtract"(%arg0, %0) : (tensor<?x?xf32>, tensor<?x?xf32>) -> tensor<?x?xf32>
+  %2 = "xla_hlo.add"(%1, %1) : (tensor<?x?xf32>, tensor<?x?xf32>) -> tensor<?x?xf32>
+  // CHECK: %[[RET:.*]]:2 = "xla_hlo.fusion"
```

Comment:
I'd likely add
```
// CHECK: xla_hlo.add
// CHECK: xla_hlo.subtract
// CHECK: xla_hlo.add
// CHECK: return %[RET]#0, %[RET]#1
```

Otherwise I don't think you can be sure that all the operations are in the fusion. You have a similar property with other tests being a bit under-constrained below.

(I'll also run some test coverage on the current version of the PR)